### PR TITLE
add Intl.supportedValuesOf

### LINKED ISSUE #49231: Add `Intl.supportedValuesOf` enumeration to `lib.*.intl.d.ts` as support in Node.js18, and v8, and various browsers exists
# lib Update Request

## Configuration Check

My compilation *target* is `ES2022` and my *lib* is `"esnext.intl", "es2021.intl", "ESNext","ES2022", "ES2021", "es6"`.

## Missing / Incorrect Definition
The static method [`Intl.supportedValuesOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/supportedValuesOf) is supported by Node.js since [v18 release from April](https://nodejs.org/en/blog/announcements/v18-release-announce/#v8-10-1). This is missing in the definition of `Intl`.

## Sample Code
```
Intl.supportedValuesOf("timeZone").forEach(function(timeZone) {
   console.log(timeZone);
});
```
## Documentation Link
[Ecmascript](https://tc39.es/proposal-intl-enumeration/#sec-intl.supportedvaluesof) - it is listed as a stage3 proposal, but given that it is already implemented in [v8](https://v8.dev/blog/v8-release-99#intl-enumeration) & mentioned in the [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/supportedValuesOf) this seems reasonable.

### PR DESCRIPTION
**Pull request by KhafraDev**
Fixes #49231

### CODE REVIEW

**Comment by KhafraDev** [comment]
@microsoft-github-policy-service agree

File: src/lib/es2022.intl.d.ts [hunk 1]
```diff
@@ -88,4 +88,17 @@ declare namespace Intl {
          */
         supportedLocalesOf(locales: BCP47LanguageTag | BCP47LanguageTag[], options?: Pick<SegmenterOptions, "localeMatcher">): BCP47LanguageTag[];
     };
+
+    type EnumerationKeys = "calendar" | "collation" | "currency" | "numberingSystem" | "timeZone" | "unit";
```
Comment (DanielRosenwasser) [inline src/lib/es2022.intl.d.ts:92]: I don't know if we should be defining a type alias for this - especially one as broadly named as `EnumerationKeys`.

Comment (KhafraDev) [reply src/lib/es2022.intl.d.ts:92]: I could inline it or just make it a string if preferred, but the proposal that added it is called [Intl enumeration](https://github.com/tc39/proposal-intl-enumeration) so I thought the name was fitting. 😄

Comment (DanielRosenwasser) [reply src/lib/es2022.intl.d.ts:92]: Yeah, I think for now let's inline it if that works.

File: src/lib/es2022.intl.d.ts [hunk 2]
```diff
@@ -88,4 +88,17 @@ declare namespace Intl {
...
     };
+
+    type EnumerationKeys = "calendar" | "collation" | "currency" | "numberingSystem" | "timeZone" | "unit";
+
+    /**
+     * Returns a list of the supported collation, calendar, currency, numbering system, timezones, and units by the implementation.
+     *
+     * @param key A string indicating the category of values to return.
+     *
+     * @returns {string[]} A list of the supported values.
```
Comment (sandersn) [inline src/lib/es2022.intl.d.ts:99]:
```suggestion
     * @returns A sorted array of the supported values.
```

**Review by sandersn — CHANGES_REQUESTED** [review]
I want to get this into 5.1, so I'll address the requests myself and push a commit.

**Comment by sandersn** [comment]
Addresses https://github.com/microsoft/TypeScript/issues/52956

**Comment by sandersn** [comment]
According [to the github repo](https://github.com/tc39/proposal-intl-enumeration), this is still at stage 3. It should go into esnext, not es2022.

**Comment by KhafraDev** [comment]
However it's marked as stage 4 in the main tc39 repo: https://github.com/tc39/proposals/blob/faa00dc13c48463a02dfe3aa6387b7082b41ce27/ecma402/finished-proposals.md

**Review by sandersn — APPROVED** [review]

**Comment by danielmarcano** [comment]
This is still throwing an error in a Next.js project with TypeScript v5.0.4:

image

Here's my tsconfig.json file:

```ts
{
  "compilerOptions": {
    /* Visit https://aka.ms/tsconfig.json to read more about this file */
    /* Language and Environment */
    "target": "es5" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ] /* Specify a set of bundled library declaration files that describe the target runtime environment. */,
    "jsx": "preserve" /* Specify what JSX code is generated. */,
    /* Modules */
    "module": "commonjs" /* Specify what module code is generated. */,
    "moduleResolution": "node" /* Specify how TypeScript looks up a file from a given module specifier. */,
    "resolveJsonModule": true /* Enable importing .json files */,
    /* JavaScript Support */
    "allowJs": true /* Allow JavaScript files to be a part of your program. Use the `checkJS` option to get errors from these files. */,
    /* Emit */
    "noEmit": true /* Disable emitting files from a compilation. */,
    /* Interop Constraints */
    "isolatedModules": true /* Ensure that each file can be safely transpiled without relying on other imports. */,
    "esModuleInterop": true /* Emit additional JavaScript to ease support for importing CommonJS modules. This enables `allowSyntheticDefaultImports` for type compatibility. */,
    "forceConsistentCasingInFileNames": true /* Ensure that casing is correct in imports. */,
    /* Type Checking */
    "strict": true /* Enable all strict type-checking options. */,
    "noImplicitAny": true /* Enable error reporting for expressions and declarations with an implied `any` type.. */,
    "strictNullChecks": true /* When type checking, take into account `null` and `undefined`. */,
    "strictFunctionTypes": true /* When assigning functions, check to ensure parameters and the return values are subtype-compatible. */,
    "strictBindCallApply": true /* Check that the arguments for `bind`, `call`, and `apply` methods match the original function. */,
    "strictPropertyInitialization": true /* Check for class properties that are declared but not set in the constructor. */,
    "noImplicitThis": true /* Enable error reporting when `this` is given the type `any`. */,
    "alwaysStrict": true /* Ensure 'use strict' is always emitted. */,
    "noUnusedLocals": true /* Enable error reporting when a local variables aren't read. */,
    "noUnusedParameters": true /* Raise an error when a function parameter isn't read */,
    "noImplicitReturns": true /* Enable error reporting for codepaths that do not explicitly return in a function. */,
    "noFallthroughCasesInSwitch": true /* Enable error reporting for fallthrough cases in switch statements. */,
    "noImplicitOverride": true /* Ensure overriding members in derived classes are marked with an override modifier. */,
    "skipLibCheck": true /* Skip type checking all .d.ts files. */,
    "incremental": true,
    "baseUrl": ".",
    "paths": {
      "$/*": ["./src/*"]
    }
  },
  "include": [
    "custom.d.ts",
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx",
    "src/styles/fonts.css",
    "@types/*.d.ts",
  ],
  "exclude": ["node_modules", "jest.setup.js"]
}

```

**Pull request merged**